File: 58. Length of Last Word.py
'''
Given a string s consisting of words and spaces, return the length of the last word in the string.

A word is a maximal
substring
consisting of non-space characters only.
'''

class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        return len(s.rstrip().split(" ")[-1])

        l = 0
        for i in range(len(s)-1,-1,-1):
            if s[i] == " " and l == 0:
                continue
            elif s[i] == " ":
                break
            else:
                l += 1
        return l

File: 58. Length of Last Word.c

/*
Given a string s consisting of words and spaces, return the length of the last word in the string.

A word is a maximal
substring
consisting of non-space characters only.
*/
int lengthOfLastWord(char* s) {
    // base case
    if(*s == '\0'){
        return 0;
    }

    //lstrip(' ')
    while(*s == ' '){
        s++;
    }

    // count the length of the current string
    int length = 0;
    while(*s != ' ' && *s != '\0'){
        length++;
        s++;
    }

    int right = lengthOfLastWord(s);
    length = right <= 0 ? length : right;
    return length;

}